tests/eos-test-utils.c

```diff
@@ -0,0 +1,1960 @@
+                      error))
+    return FALSE;
+
+  unix_utc = g_key_file_get_int64 (keyfile,
+                                   "main",
+                                   "UnixUTCTimestamp",
+                                   &local_error);
+  if (local_error != NULL)
+    {
+      g_propagate_error (error, g_steal_pointer (&local_error));
+      return FALSE;
+    }
+
+  timestamp = g_date_time_new_from_unix_utc (unix_utc);
+  if (timestamp == NULL)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "Invalid UnixUTCTimestamp");
+      return FALSE;
+    }
+
+  *out_timestamp = g_steal_pointer (&timestamp);
+  return TRUE;
+}
+
+static GKeyFile *
+generate_definition (GFile *client_root,
+                     guint16 port,
+                     GDateTime *timestamp,
+                     const gchar *ostree_path)
+{
+  SSDEF;
+  GKeyFile *definition = g_key_file_new ();
+  g_autofree gchar *basename = g_file_get_basename (client_root);
+  g_auto(GStrv) txt = NULL;
+
+  g_key_file_set_string (definition,
+                         "service",
+                         "name",
+                         SS (g_strdup_printf ("Test Update Server at %s",
+                                              basename)));
+
+  g_key_file_set_string (definition,
+                         "service",
+                         "domain",
+                         SS (g_strdup_printf ("%s.local",
+                                              basename)));
+
+  g_key_file_set_string (definition,
+                         "service",
+                         "address",
+                         "127.0.0.1");
+
+  g_key_file_set_integer (definition,
+                          "service",
+                          "port",
+                          port);
+
+  txt = generate_strv (SS (envvar ("eos_txt_version",
+                                   "2")),
+                       SS (envvar ("eos_branch_file_timestamp",
+                                   SS (g_strdup_printf ("%" G_GINT64_FORMAT,
+                                                        g_date_time_to_unix (timestamp))))),
+                       SS (envvar ("eos_ostree_path",
+                                   ostree_path)),
+                       NULL);
+
+  g_key_file_set_string_list (definition,
+                              "service",
+                              "txt",
+                              (const gchar *const *) txt,
+                              g_strv_length (txt));
+
+  return definition;
+}
+
+static GFile *
+update_server_quit_file (GFile *update_server_dir)
+{
+  return g_file_get_child (update_server_dir, "quit-file");
+}
+
+static gboolean
+prepare_update_server_dir (GFile *update_server_dir,
+                           GError **error)
+{
+  g_autoptr(GFile) quit_file = NULL;
+
+  if (!create_directory (update_server_dir, error))
+    return FALSE;
+
+  quit_file = update_server_quit_file (update_server_dir);
```

Comment:
Maybe rename this function to `server_quit_file_from_server_dir` - I initially read this as "update the server quit file"